core/heap.c

```diff
@@ -1401,6 +1401,9 @@ vmm_heap_init()
     if (DYNAMO_OPTION(vm_reserve)) {
         vmm_heap_unit_init(&heapmgt->vmheap, DYNAMO_OPTION(vm_size));
     }
+    dynamo_vm_areas_lock();
+    heap_exiting = false;
```

Comment:
Hmm, the approach for #2157 has been the opposite, to pay a cost only when exiting on a detach.  It may seem uglier, but init time does matter: we want to support running shell scripts and other sequences with a series of rapid process creation and teardown events.  Furthermore, explicit initialization of globals causes extra overhead beyond just a memory store by incurring copy-on-write costs (.bss is part of the shared image on Windows).

We also have an invariant where our process init routines can avoid the cost of locks: only one thread will be in DR during process init.

If this were "regular" code I would agree that this is cleaner, but I think for DR we should avoid unnecessary work.  I'm sure this single lock and store won't be measurable by itself but if we did this for every static var it will likely have a noticeable impact.